game-core/src/test/java/games/strategy/engine/message/RemoteActionCodeTest.java

```diff
@@ -0,0 +1,49 @@
+package games.strategy.engine.message;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.lang.reflect.Method;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.aggregator.AggregateWith;
+import org.junit.jupiter.params.aggregator.ArgumentsAccessor;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregationException;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregator;
+import org.junit.jupiter.params.provider.CsvFileSource;
+
+public class RemoteActionCodeTest {
+
+  @ParameterizedTest
+  @CsvFileSource(resources = "/required-op-codes.txt")
+  void verifyCorrectOpCode(
+      final int opCode, @AggregateWith(MethodAggregator.class) final Method method) {
+    final var remoteActionCode = method.getAnnotation(RemoteActionCode.class);
```

Comment:
`var` is perhaps confusing here, I'm not offhand 100% sure what the return type is of `getAnnotation`. I think it is probably `Class`, but not fully evident. Perhaps if the variable name were `remoteActionCodeClass`, it would be clear and would not benefit from type information.